modules/login/src/main/resources/META-INF/resources/login/js/registration.js

```diff
@@ -0,0 +1,401 @@
+var registration = function($) {
```

Comment:
This is a tough one since the code doesn't use a particular 'unit-testable' framework like angular. If you have ideas about how we can add regular unit testing to the login module jquery code then I'd love to discuss.